web/src/main/java/org/cbioportal/web/StructuralVariantController.java

```diff
@@ -58,26 +58,26 @@
             consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Fetch structural variants for geneticProfileIds, hugoGeneSymbols and also sampleIdentifiers")
     public ResponseEntity<List<StructuralVariant>> fetchStructuralVariants(
-            @ApiParam(required = true, value = "List of geneticProfileStableIds, hugoGeneSymbols and sampleIdentifiers")
+            @ApiParam(required = true, value = "List of hugoGeneSymbols and molecularProfileId or sampleMolecularIdentifiers")
             @Valid @RequestBody StructuralVariantFilter structuralVariantFilter) {
         
         List<StructuralVariant> structuralVariantList;
-        
-        if (structuralVariantFilter.getSampleIdentifiers() != null) {
-            List<SampleIdentifier> sampleIdentifiers = structuralVariantFilter.getSampleIdentifiers();
-            List<String> studyIds = new ArrayList<>();
+
+        if (structuralVariantFilter.getSampleMolecularIdentifiers() != null) {
+            List<SampleMolecularIdentifier> sampleMolecularIdentifiers = structuralVariantFilter.getSampleMolecularIdentifiers();
+            List<String> molecularProfileIds = new ArrayList<>();
             List<String> sampleIds = new ArrayList<>();
 
-            for (SampleIdentifier sampleIdentifier : sampleIdentifiers) {
-                studyIds.add(sampleIdentifier.getStudyId());
-                sampleIds.add(sampleIdentifier.getSampleId());
+            for (SampleMolecularIdentifier sampleMolecularIdentifier : sampleMolecularIdentifiers) {
+                molecularProfileIds.add(sampleMolecularIdentifier.getMolecularProfileId());
+                sampleIds.add(sampleMolecularIdentifier.getSampleId());
             }
-            structuralVariantList = structuralVariantService.fetchStructuralVariants(structuralVariantFilter.getGeneticProfileStableIds(), structuralVariantFilter.getHugoGeneSymbols(), studyIds, sampleIds);
+            structuralVariantList = structuralVariantService.fetchStructuralVariants(molecularProfileIds, structuralVariantFilter.getHugoGeneSymbols(), sampleIds);
             
         } else {
-            List<String> studyIds = new ArrayList<>();
+            List<String> molecularProfileIds = new ArrayList<>();
             List<String> sampleIds = new ArrayList<>();
-            structuralVariantList = structuralVariantService.fetchStructuralVariants(structuralVariantFilter.getGeneticProfileStableIds(), structuralVariantFilter.getHugoGeneSymbols(), studyIds, sampleIds);
+            structuralVariantList = structuralVariantService.fetchStructuralVariants(molecularProfileIds, structuralVariantFilter.getHugoGeneSymbols(), sampleIds);
```

Comment:
Why using empty array? Shouldn't this use structuralVariantFilter.getMolecularProfileIds()?

Reply:
You are correct! This is a mistake, should indeed be structuralVariantFilter.getMolecularProfileIds().